pkg/component/etcd/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -0,0 +1,175 @@
+		labels[k] = v
+	}
+
+	return labels
+}
+
+// getAnnotations returns the annotations for the PDB object
+func getAnnotations(val *Values) map[string]string {
+	annotations := map[string]string{
+		ownedByAnnotationKey:   fmt.Sprintf("%s/%s", val.EtcdNameSpace, val.EtcdName),
+		ownerTypeAnnotationKey: "etcd",
+	}
+	return annotations
+}
+
+// emptyPodDisruptionBudget returns an empty PDB object with only the name and namespace as part of the object meta
+func (c *component) emptyPodDisruptionBudget(name string, k8sversion *semver.Version) interface{} {
+	fmt.Println("In PDB emptyPDB")
+	if k8sutils.ConstraintK8sGreaterEqual121.Check(k8sversion) {
+		return &policyv1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: c.namespace,
+			},
+		}
+	} else {
+		return &policyv1beta1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: c.namespace,
+			},
+		}
+	}
+}
+
+func (c *component) Deploy(ctx context.Context) error {
+	fmt.Println("In PDB deploy")
+	k8sversion, _ := utils.GetTargetK8sVersion(c.controllerConfig)
+	pdb := c.emptyPodDisruptionBudget(c.values.EtcdName, k8sversion)
+
+	if err := c.syncPodDisruptionBudget(ctx, pdb); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *component) syncPodDisruptionBudget(ctx context.Context, pdb interface{}) error {
+	switch pdb := pdb.(type) {
+	case *policyv1.PodDisruptionBudget:
+		_, err := controllerutils.GetAndCreateOrMergePatch(ctx, c.client, pdb, func() error {
+			pdb.Annotations = getAnnotations(c.values)
+			pdb.Labels = getLabels(c.values)
+			pdb.OwnerReferences = getOwnerReferences(c.values)
+			pdb.Spec.MinAvailable = &intstr.IntOrString{
+				IntVal: c.values.MinAvailable,
+				Type:   intstr.Int,
+			}
+			pdb.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: getLabels(c.values),
+			}
+			return nil
+		})
+		return err
+	case *policyv1beta1.PodDisruptionBudget:
+		_, err := controllerutils.GetAndCreateOrMergePatch(ctx, c.client, pdb, func() error {
+			pdb.Annotations = getAnnotations(c.values)
+			pdb.Labels = getLabels(c.values)
+			pdb.OwnerReferences = getOwnerReferences(c.values)
+			pdb.Spec.MinAvailable = &intstr.IntOrString{
+				IntVal: c.values.MinAvailable,
+				Type:   intstr.Int,
+			}
+			pdb.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: getLabels(c.values),
+			}
+			return nil
+		})
+		return err
+	}
+	return nil
+}
+
+func (c *component) Destroy(ctx context.Context) error {
+	k8sversion, _ := utils.GetTargetK8sVersion(c.controllerConfig)
+	pdb := c.emptyPodDisruptionBudget(c.values.EtcdName, k8sversion)
+	switch pdb.(type) {
+	case *policyv1.PodDisruptionBudgetList:
+		pdbv1 := pdb.(*policyv1.PodDisruptionBudget)
+		return client.IgnoreNotFound(c.client.Delete(ctx, pdbv1))
+	case *policyv1beta1.PodDisruptionBudgetList:
+		pdbv1beta1 := pdb.(*policyv1.PodDisruptionBudget)
+		return client.IgnoreNotFound(c.client.Delete(ctx, pdbv1beta1))
+	}
```

Comment:
```suggestion
pdb := c.emptyPodDisruptionBudget(c.values.EtcdName, k8sversion)
return client.IgnoreNotFound(c.client.Delete(ctx, pdb))
```
`client.Delete` takes `client.Object` as argument, you can directly pass the pdb.

Reply:
`emptyPodDisruptionBudget` returns an `interface{}` and I can't pass that to `Delete(..)`

Reply:
Why not return `client.Object` from `emptyPodDisruptionBudget `?

Reply:
Thanks for pointing this out
Updated to `emptyPodDisruptionBudget` to return `client.Object`